hoodie-common/src/main/java/com/uber/hoodie/common/util/FSUtils.java

```diff
@@ -49,8 +50,17 @@
     // Log files are of this pattern - b5068208-e1a4-11e6-bf01-fe55135034f3_20170101134598.avro.delta.1
     private static final Pattern LOG_FILE_PATTERN = Pattern.compile("(.*)_(.*)\\.(.*)\\.(.*)\\.([0-9]*)");
     private static final int MAX_ATTEMPTS_RECOVER_LEASE = 10;
+    private static FileSystem fs;
+
+    @VisibleForTesting
+    public static void setFs(FileSystem fs) {
+        FSUtils.fs = fs;
```

Comment:
I understand you need control over the filesystem initialization, but this seems a little ugly.. I actually prefer if you can mock or stub FSUtils.getFs() for the purposes of the test.. is that doable?